shell/platform/android/io/flutter/embedding/android/FlutterFragment.java

```diff
@@ -470,7 +473,26 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
     Log.v(TAG, "Creating FlutterView.");
     flutterView = new FlutterView(getActivity(), getRenderMode(), getTransparencyMode());
     flutterView.addOnFirstFrameRenderedListener(onFirstFrameRenderedListener);
-    return flutterView;
+
+    flutterSplashView = new FlutterSplashView(getContext());
+    // TODO(mattcarroll): Find a better solution to this ID. This is a random, static ID.
+    // It might conflict with other Views, and it means that only a single FlutterSplashView
+    // can exist in a View hierarchy at one time.
+    flutterSplashView.setId(486947586);
```

Comment:
It needs an ID so that Android will allow it to save state. Normally that ID would be assigned in a layout file, or at least assigned an R.id.something, but we don't support layout inflation or ID resource declarations in the engine.

That's a great point on `View#generateId()`, I didn't know that existed. I'll use that on APIs where its available.